engine/src/main/scala/cromwell/services/ServiceRegistryActor.scala

```diff
@@ -0,0 +1,51 @@
+package cromwell.services
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.{ConfigFactory, Config, ConfigObject}
+import cromwell.engine.db.DataAccess
+import cromwell.services.ServiceRegistryActor._
+import lenthall.config.ScalaConfig._
+
+import scala.collection.JavaConverters._
+
+object ServiceRegistryActor {
+  case class ServiceRegistryFailure(serviceName: String)
+  trait ServiceRegistryMessage {
+    def serviceName: String
+  }
+
+  def props(dataAccess: DataAccess, config: Config) = {
+    Props(ServiceRegistryActor(dataAccess, config))
+  }
+}
+
+case class ServiceRegistryActor(dataAccess: DataAccess, globalConfig: Config) extends Actor {
```

Comment:
Not that I disagree... I just think we should not over-engineer this a whole bunch and let it evolve a little more...  is this sort of what you want?

```scala
object ServiceRegistryActor {
  case class ServiceRegistryFailure(serviceName: String)
  trait ServiceRegistryMessage {
    def serviceName: String
  }

  def props(config: Config) = {
    Props(ServiceRegistryActor(config))
  }

  def serviceProps(globalConfig: Config): Map[String, Props] = {
    val serviceNamesToConfigStanzas = globalConfig.getObject("services").entrySet.asScala.map(x => x.getKey -> x.getValue).toMap
    serviceNamesToConfigStanzas map {
      case (serviceName, config: ConfigObject) => serviceName -> makeServiceProps(serviceName, globalConfig, config.toConfig)
      case (serviceName, _) => throw new Exception(s"Invalid configuration for service $serviceName")
    }
  }

  private def makeServiceProps(serviceName: String, globalConfig: Config, serviceStanza: Config): Props = {
    val serviceConfigStanza = serviceStanza.getConfigOption("config").getOrElse(ConfigFactory.parseString(""))
    val className = serviceStanza.getStringOr(
      "class",
      throw new Exception(s"Invalid configuration for service $serviceName: missing 'class' definition")
    )
    val method = Class.forName(className).getMethod("props", classOf[DataAccess], classOf[Config], classOf[Config])
    method.invoke(None, DataAccess.globalDataAccess, serviceConfigStanza, globalConfig).asInstanceOf[Props]
  }
}

case class ServiceRegistryActor(globalConfig: Config) extends Actor {
  import ServiceRegistryActor._

  val services: Map[String, ActorRef] = serviceProps(globalConfig) map {
    case (name, props) => name -> context.actorOf(props)
  }

  def receive = {
    case msg: ServiceRegistryMessage =>
      services.get(msg.serviceName) match {
        case Some(ref) => ref.tell(msg, sender)
        case None => sender ! ServiceRegistryFailure(msg.serviceName)
      }
  }
}
```